src/microns_to_nwb/ophys/micronstiffimagingextractor.py

```diff
@@ -0,0 +1,89 @@
+from typing import Tuple, Optional
+
+import numpy as np
+import zarr
+from fsspec import filesystem
+from fsspec.implementations.cached import CachingFileSystem
+from neuroconv.utils import FilePathType, FloatType
+from roiextractors import ImagingExtractor
+from tifffile import imread
+
+
+class MicronsTiffImagingExtractor(ImagingExtractor):
+    extractor_name = "MicronsTiffImaging"
+    is_writable = True
+    mode = "file"
+
+    def __init__(
+        self,
+        file_path: FilePathType,  # S3 file path to TIF file
+        cache_storage: FilePathType,  # local cache storage for fsspec
+        sampling_frequency: FloatType,
+        plane_index: int,  # the field index
+        num_frames_per_plane: int,
+    ):
+        self._s3_file = None
+        self.file_path = file_path
+        super().__init__()
+
+        self._sampling_frequency = sampling_frequency
+        self._plane_index = plane_index
+        self._num_frames = num_frames_per_plane
+
+        # Lazy loading of image data from S3
+        self._video = self.open(cache_storage=cache_storage)
+
+        assert self._video.shape[0] % self._num_frames == 0
+        self._num_planes = int(self._video.shape[0] / self._num_frames)
+        self._frames_range = np.arange(self._plane_index, self._num_planes * self._num_frames, self._num_planes)
+
+        self._num_channels = 1
+        self._num_rows, self._num_columns = self._video.shape[1:]
+
+    def open(self, cache_storage):
+        cfs = CachingFileSystem(
+            fs=filesystem("s3", anon=True),
+            cache_storage=cache_storage,  # Local folder for the cache
+        )
+
+        self._s3_file = cfs.open(self.file_path, "rb")
+        zarr_store = imread(self._s3_file, aszarr=True)
+
+        return zarr.open(zarr_store, mode="r")
+
+    def close(self):
+        self._s3_file.close()
+
+    def get_frames(self, frame_idxs, channel: int = 0):
+        return self._video.oindex[self._frames_range[frame_idxs], :, :]
+
+    def get_video(self, start_frame=None, end_frame=None, channel: Optional[int] = 0):
```

Comment:
Definitely shouldn't need `channel` here: https://github.com/catalystneuro/ahrens-lab-to-nwb/blob/main/src/ahrens_lab_to_nwb/yu_mu_cell_2019/extractors/yu_mu_cell_2019_imaging_extractor.py#L41worked fine without it

Reply:
Great I'll fix it, thanks for checking!

Reply:
The 'here' in this comment is for `get_video`, which is the only argument I overrode. Also the only one you should need - you might not actually have to override `get_frames` at all in your extractor.

For a while now `get_frames` has been written in terms of `get_video`: https://github.com/catalystneuro/roiextractors/blob/a902783dfad226779110e2341b96ae92107f2b6e/src/roiextractors/imagingextractor.py#L62-L67

Reply:
I understand that, what I meant is when I use stubbing with `FrameSliceImagingExtractor` it expects that the parent have a channel arg in `get_video` (also in `get_frames`):
https://github.com/catalystneuro/roiextractors/blob/a902783dfad226779110e2341b96ae92107f2b6e/src/roiextractors/imagingextractor.py#L221

Reply:
I see - I don't think this came up for me due to my use of a `MultiImagingExtractor`